presto-main/src/test/java/com/facebook/presto/block/AbstractTestBlock.java

```diff
@@ -224,6 +225,18 @@ private void assertBlockSize(Block block)
         long expectedSecondHalfSize = copyBlockViaBlockSerde(secondHalf).getSizeInBytes();
         assertEquals(secondHalf.getSizeInBytes(), expectedSecondHalfSize);
         assertEquals(block.getRegionSizeInBytes(firstHalf.getPositionCount(), secondHalf.getPositionCount()), expectedSecondHalfSize);
+
+        boolean[] positions = new boolean[block.getPositionCount()];
+        for (int i = 0; i < firstHalf.getPositionCount(); ++i) {
+            positions[i] = true;
+        }
+        assertEquals(firstHalf.getSizeInBytes(), block.getPositionsSizeInBytes(positions));
+        fill(positions, true);
+        assertEquals(block.getSizeInBytes(), block.getPositionsSizeInBytes(positions));
+        for (int i = 0; i < firstHalf.getPositionCount(); ++i) {
+            positions[i] = false;
+        }
+        assertEquals(secondHalf.getSizeInBytes(), block.getPositionsSizeInBytes(positions));
```

Comment:
```
        boolean[] positions = new boolean[block.getPositionCount()];
        Arrays.fill(positions, true);
        assertEquals(block.getPositionsSizeInBytes(positions), expectedBlockSize);
        Arrays.fill(positions, false);
        Arrays.fill(positions, 0, firstHalf.getPositionCount(), true);
        assertEquals(block.getPositionsSizeInBytes(positions), expectedFirstHalfSize);
        Arrays.fill(positions, false);
        Arrays.fill(positions, firstHalf.getPositionCount(), block.getPositionCount(), true);
        assertEquals(block.getPositionsSizeInBytes(positions), expectedSecondHalfSize);

Reply:
May I ask why you'd suggest writing it this way (other than the for loop is not needed)? They are testing the same things, just in different order, right?